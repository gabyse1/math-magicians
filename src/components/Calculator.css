.calc__container {
  width: 100%;
  max-width: 280px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(6, 1fr);
}

.grid-col-span-4 {
  grid-column: span 4;
}

#calc__btn-0 {
  grid-column: span 2;
}

.btn {
  padding: 0.5rem;
  background: #dee2e6;
  border: 1px solid #adb5bd;
  font-family: 'Signika', sans-serif;
  font-weight: 400;
  color: #212529;
  cursor: pointer;
}

.btn:hover {
  background: radial-gradient(#dee2e6, #ced4da, #adb5bd 90%);
  font-weight: bold;
}

.bg-accent {
  font-size: 1rem;
}

.calc__result {
  padding: 0.5rem;
  font-size: 1.5rem;
  color: #fff;
  text-align: right;
  background-color: #495057;
}
