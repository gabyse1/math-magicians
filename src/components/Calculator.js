import React, { useState } from 'react';
import '../scss/Calculator.scss';
import BtnComponent from './BtnComponent';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calcObject, setCalcObject] = useState({
    total: null, next: null, operation: null, display: '0',
  });

  const calculateHandler = (btnName) => {
    const obj = calculate(calcObject, btnName);
    setCalcObject(obj);
  };

  return (
    <main className="page__calculator">
      <div data-testid="calc__description" className="calc__description">
        <h2 className="main__title">Let&apos;s do some Math</h2>
      </div>
      <div data-testid="calc__box" className="calc__box">
        <div data-testid="calc__display-box" className="calc__result grid-col-span-4">{calcObject.display}</div>
        <BtnComponent values={['ac', 'AC']} calculateHandler={calculateHandler} />
        <BtnComponent values={['addsub', '+/-']} calculateHandler={calculateHandler} />
        <BtnComponent values={['percen', '%']} calculateHandler={calculateHandler} />
        <BtnComponent values={['divisi', 'รท']} calculateHandler={calculateHandler} />
        <BtnComponent values={['7', '7']} calculateHandler={calculateHandler} />
        <BtnComponent values={['8', '8']} calculateHandler={calculateHandler} />
        <BtnComponent values={['9', '9']} calculateHandler={calculateHandler} />
        <BtnComponent values={['multip', 'x']} calculateHandler={calculateHandler} />
        <BtnComponent values={['4', '4']} calculateHandler={calculateHandler} />
        <BtnComponent values={['5', '5']} calculateHandler={calculateHandler} />
        <BtnComponent values={['6', '6']} calculateHandler={calculateHandler} />
        <BtnComponent values={['substr', '-']} calculateHandler={calculateHandler} />
        <BtnComponent values={['1', '1']} calculateHandler={calculateHandler} />
        <BtnComponent values={['2', '2']} calculateHandler={calculateHandler} />
        <BtnComponent values={['3', '3']} calculateHandler={calculateHandler} />
        <BtnComponent values={['add', '+']} calculateHandler={calculateHandler} />
        <BtnComponent values={['0', '0']} calculateHandler={calculateHandler} />
        <BtnComponent values={['dot', '.']} calculateHandler={calculateHandler} />
        <BtnComponent values={['equal', '=']} calculateHandler={calculateHandler} />
      </div>
    </main>
  );
};

export default Calculator;
