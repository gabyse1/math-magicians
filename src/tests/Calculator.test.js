import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('<Calculator /> component', () => {
  it('Component render correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Calculator box exist', () => {
    const { getByTestId } = render(<Calculator />);
    const calcBox = getByTestId('calc__box');
    expect(calcBox).toBeDefined();
  });

  it('Calculator has 19 buttons as children', () => {
    const { getAllByTestId } = render(<Calculator />);
    const arrBtns = getAllByTestId('calcBtn');
    expect(arrBtns.length).toBe(19);
  });
});
