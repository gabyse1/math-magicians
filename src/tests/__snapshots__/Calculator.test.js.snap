// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator /> component Component render correctly 1`] = `
<main
  className="page__calculator"
>
  <div
    className="calc__description"
  >
    <h2
      className="main__title"
    >
      Let's do some Math
    </h2>
  </div>
  <div
    className="calc__box"
    data-testid="calc__box"
  >
    <div
      className="calc__result grid-col-span-4"
    >
      0
    </div>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-ac"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-addsub"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-percen"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-divisi"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-multip"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-substr"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-add"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn"
      data-testid="calcBtn"
      id="calc__btn-equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</main>
`;
